invenio_records_restapi/restful.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""REST API resources."""
+
+from flask import jsonify, make_response, request, url_for
+
+from flask.ext.restful import Resource, abort
+
+from invenio_db import db
+
+from invenio_records.api import Record
+
+from jsonpatch import JsonPatchException, JsonPointerException
+
+from sqlalchemy.orm.exc import NoResultFound
+
+
+def record_self_link(record, **kwargs):
+    """Creates self link to a given record.
+
+    :Parameters:
+        - `record` (Record): record to which the generated link will point.
+        - `**kwargs`: additional parameters given to flask.url_for.
+    :Returns: link pointing to the given record.
+    :Returns Type: str
+    """
+    # '.' is needed to reference the current blueprint context
+    return url_for('.' + RecordAPI.api_endpoint,
+                   record_id=record.model.id, **kwargs)
+
+
+def request_wants_json():
+    """Test if the request accept type accepts 'application/json'.
+    :Returns: True if the request's accept type contains 'application/json'
+    else False.
+    :Returns Type: bool
+    """
+    best = request.accept_mimetypes \
+        .best_match(['application/json'])
+    return best == 'application/json'
+
+
+def record_to_response(record):
```

Comment:
I just reported some issues to flask-restful about their *representation functions* flask-restful/flask-restful#520, flask-restful/flask-restful#521. I will use `Resource.representations` for the time being but it does not seem to be documented so it might change later on.